### PR TITLE
SASS-9711:Allowances section: Create 'Allowances' Hyperlink and Allow…

### PR DESCRIPTION
**Pull request by RGlossop**
…ances Intro Pages

### CODE REVIEW

**Review by vivekhmrc — DISMISSED** [review]

File: app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                                       val controllerComponents: MessagesControllerComponents,
+                                       view: ForeignPropertyAllowancesStartView,
+                                       businessService: BusinessService
+                                     )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with PropertyDetailsHandler {
+
+  def onPageLoad(taxYear: Int, countryCode: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      val countryName = request.userAnswers.get(IncomeSourceCountries).map(_.array.toList).flatMap(countryList =>
+        countryList.find { country => country.code == countryCode
+        }).get.name
```
Comment (tapiwa-tiyemba) [inline app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala:51]: Hey @RGlossop you shouldn't use .get on an option. Please update this.

File: app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+      val hc = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+      withUkPropertyDetails[Result](businessService, request.user.nino, request.user.mtditid) {
+        (propertyData: PropertyDetails) =>
+          Future(
+            Ok(view(taxYear, countryCode, countryName, request.user.isAgentMessageKey, propertyData.accrualsOrCash.get))
```
Comment (tapiwa-tiyemba) [inline app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala:57]: We need to properly handle the case where accrualsOrCash is None

File: app/controllers/foreign/expenses/ForeignPropertyExpensesStartController.scala [hunk 3]
```diff
@@ -38,10 +38,10 @@ class ForeignPropertyExpensesStartController @Inject()(
 
   def onPageLoad(taxYear:Int, countryCode: String): Action[AnyContent] = (identify andThen getData andThen requireData) {
     implicit request =>
-      val countryName = request.userAnswers.get(IncomeSourceCountries).map(_.array.toList).flatMap( countryList =>
-        countryList.find{ country => country.code == countryCode
-        }).get.name
+      val maybeCountryName = request.userAnswers.get(IncomeSourceCountries).flatMap(_.find(_.code==countryCode)).map(_.name)
+      val countryName = maybeCountryName.getOrElse("")
       val income = request.userAnswers.get(ForeignPropertyRentalIncomePage(countryCode))
+
```
Comment (tapiwa-tiyemba) [inline app/controllers/foreign/expenses/ForeignPropertyExpensesStartController.scala:44]: you need a case for when income is None

File: app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+  def onPageLoad(taxYear: Int, countryCode: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      val maybeCountryName = request.userAnswers.get(IncomeSourceCountries).flatMap(_.find(_.code==countryCode)).map(_.name)
+      val countryName = maybeCountryName.getOrElse("")
+
+      val hc = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
+      withUkPropertyDetails[Result](businessService, request.user.nino, request.user.mtditid) {
+        (propertyData: PropertyDetails) =>
+          propertyData.accrualsOrCash.getOrElse(None) match {
```
Comment (tapiwa-tiyemba) [inline app/controllers/foreign/allowances/ForeignPropertyAllowancesStartController.scala:55]: same here

**Review by tapiwa-tiyemba — CHANGES_REQUESTED** [review]

**Review by tapiwa-tiyemba — APPROVED** [review]

**Pull request merged**